pkg/controller/certificates/authority/authority.go

```diff
@@ -78,16 +68,19 @@ func (ca *CertificateAuthority) Sign(crDER []byte, policy SigningPolicy) ([]byte
 		PublicKey:          cr.PublicKey,
 		Extensions:         cr.Extensions,
 		ExtraExtensions:    cr.ExtraExtensions,
-		NotBefore:          nbf,
 	}
-	if err := policy.apply(tmpl); err != nil {
+	if err := policy.apply(tmpl, usages); err != nil {
 		return nil, err
 	}
 
+	if !tmpl.NotBefore.Before(ca.Certificate.NotAfter) {
+		return nil, fmt.Errorf("the signer has expired: NotAfter=%v", ca.Certificate.NotAfter)
+	}
+
 	if !tmpl.NotAfter.Before(ca.Certificate.NotAfter) {
 		tmpl.NotAfter = ca.Certificate.NotAfter
 	}
-	if !now.Before(ca.Certificate.NotAfter) {
+	if !time.Now().Before(ca.Certificate.NotAfter) {
 		return nil, fmt.Errorf("refusing to sign a certificate that expired in the past")
 	}
```

Comment:
If we want to keep this check, I think I should pass in the CA into the `SigningPolicy.apply` call (to keep all the `now` logic in one place).

Reply:
Or something like:

```go
type SigningPolicy interface {
	apply(template *x509.Certificate, usages []capi.KeyUsage, caNotAfter time.Time) error
}
```